src/IP.php

```diff
@@ -115,13 +115,19 @@ public function numeric($base = 10)
     /**
      * Return binary string representation.
      *
-     * @todo could be optimized with pack() instead?
-     *
      * @return string Binary string
      */
     public function binary()
     {
-        return inet_pton($this->humanReadable());
+        $padLength = 8;
+
+        if ($this instanceof IPv6) {
+            $padLength = 32;
```

Comment:
Could padLength be a class constant (or derived from a class constant), so we avoid this test? I don't like the parent class checking if it's a child class.

Reply:
As you commented earlier, I think the static::NB_BYTES is the way to go here.